src/chain/validators-impl.ts

```diff
@@ -333,6 +333,11 @@ export class ValidatorsImpl<Chain> implements Validators<Chain> {
     return this.addStandardValidation(validator.isWhitelisted, chars);
   }
   matches(pattern: RegExp | string, modifiers?: string) {
-    return this.addStandardValidation(validator.matches, pattern, modifiers);
+    return this.addStandardValidation.apply(this, [
+      validator.matches,
+      ...(typeof pattern === 'string'
+        ? [pattern, modifiers]
+        : [pattern.source, ...[new Set((modifiers || '') + pattern.flags)].join('')]),
+    ]);
```

Comment:
Just a super nitpick, can we use a ternary expression or an if statement, instead of the apply? That's definitely correct but makes the code harder to read for new developers.

Reply:
And can you add a couple tests to check that the flags added after the regex are effectively passed to the `addStandatdValidation` function?

Reply:
No problem for the tests,
But removing the apply would mean duplicate entirely the call to addStandardValidation...
I'd say that new developers shouldn't be supposed to touch such a piece of code!
I was reluctant myself despite several years of experience! 😅
Also remember that if they fix validator library we can revert this change! 😉

Reply:
That's a nice opinion and I'm fine to keep it as is, I'll open a PR and push to get that merged in asap!

Reply:
> And can you add a couple tests to check that the flags added after the regex are effectively passed to the addStandatdValidation function?

An even better testing way would be to reproduce the same regex in the issue, calling the method twice and asserting that the validation is passed both times.

Reply:
Ok I'll add the tests tomorrow!
Thanks!!

Reply:
Great! Thank you!

Reply:
Good thing you asked for the tests...The spread was misplaced...